src/librustc/middle/def_id.rs

```diff
@@ -61,12 +61,16 @@ impl fmt::Debug for DefId {
         // Unfortunately, there seems to be no way to attempt to print
         // a path for a def-id, so I'll just make a best effort for now
         // and otherwise fallback to just printing the crate/node pair
-        try!(ty::tls::with_opt(|opt_tcx| {
-            if let Some(tcx) = opt_tcx {
-                try!(write!(f, " => {}", tcx.item_path_str(*self)));
-            }
-            Ok(())
-        }));
+        if self.is_local() { // (1)
```

Comment:
@arielb1 

> Does this fix RUST_LOG=rustc::middle::traits?

Yes, I expect so. I think I was having problems with `RUST_LOG=rustc_typeck`, but I suspect it's the same issue. And yes, a fallible version of `item_path_str` would be better.